include/pybind11/detail/common.h

```diff
@@ -229,6 +244,9 @@
 #    include <version>
 #  endif
 #endif
+#ifdef PYBIND11_HAS_STRING_VIEW
+#  include <string_view>
```

Comment:
Minor nit: To better follow an IWYU model, we should really only add the includes where they are needed in `pytypes.h and cast.h`